### ISSUE TITLE
Pre-built binaries not found for microtime@2.1.4

### ISSUE THREAD

**Issue by tforssander** [wadey/node-microtime#50]
When installing the latest node-microtime package using:

`npm install microtime` the following error message appears:

```bash
node-pre-gyp http GET https://github.com/wadey/node-microtime/releases/download/v2.1.4/node-v48-linux-x64.tar.gz
node-pre-gyp http 404 https://github.com/wadey/node-microtime/releases/download/v2.1.4/node-v48-linux-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/wadey/node-microtime/releases/download/v2.1.4/node-v48-linux-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for microtime@2.1.4 and node@6.9.5 (node-v48 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/wadey/node-microtime/releases/download/v2.1.4/node-v48-linux-x64.tar.gz
```

Have the pre-built binaries been renamed or removed?

Thanks!

**Comment by wadey** [comment]
I only made pre-built binaries for Windows, as a lot of users with Windows don't have a proper build environment. Linux and macOS users usually have no problem building the module. Did your system have any troubles building from source?

I'll see if I can hide that error message for Linux and macOS until I can make Travis prebuild a binary for them.

**Comment by tforssander** [comment]
Ah, good to know. 

I've should have check the logs better since it looks like it is actually building from source. Thanks for the quick reply!

**Comment by xwilders** [comment]
My docker builds have begun failing due to this:
> microtime@2.1.4 install /opt/meteor/dist/bundle/programs/server/npm/node_modules/microtime > node-pre-gyp install --fallback-to-build --loglevel http

> module.js:327 throw err; ^ Error: Cannot find module '../' at Function.Module._resolveFilename (module.js:325:15) at Function.Module._load (module.js:276:25) at Module.require (module.js:353:17) at require (internal/module.js:12:17) at Object.<anonymous> (/opt/meteor/dist/bundle/programs/server/npm/node_modules/microtime/node_modules/.bin/node-pre-gyp:15:20) at Module._compile (module.js:409:26) at Object.Module._extensions..js (module.js:416:10) at Module.load (module.js:343:32) at Function.Module._load (module.js:300:12) at Function.Module.runMain (module.js:441:10)

**Comment by wadey** [comment]
> My docker builds have begun failing due to this:

@xwilders Can you try now with version 2.1.5 that I just released? Fixed an issue with bundling the dependency.

**Comment by xwilders** [comment]
Thanks! It works again :)

**Comment by dsnoeck** [comment]
Hi have a similar issue on my pipeline system:
```
13:34:01.974 > microtime@2.1.5 install /home/xxx/xxx/xxx/node_modules/microtime
13:34:01.974 > node-pre-gyp install --fallback-to-build --loglevel http
13:34:01.974 
13:34:01.974 node-pre-gyp http GET https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v48-linux-x64.tar.gz
13:36:09.325 node-pre-gyp ERR! Tried to download(undefined): https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v48-linux-x64.tar.gz 
13:36:09.325 node-pre-gyp ERR! Pre-built binaries not found for microtime@2.1.5 and node@6.9.1 (node-v48 ABI) (falling back to source compile with node-gyp) 
13:36:09.325 node-pre-gyp http connect ETIMEDOUT 192.30.253.113:443 
13:36:09.325 gyp http GET https://nodejs.org/download/release/v6.9.1/node-v6.9.1-headers.tar.gz
13:38:16.728 gyp WARN install got an error, rolling back install
13:38:16.729 gyp ERR! configure error 
13:38:16.729 gyp ERR! stack Error: connect ETIMEDOUT 104.20.22.46:443
13:38:16.729 gyp ERR! stack     at Object.exports._errnoException (util.js:1026:11)
13:38:16.729 gyp ERR! stack     at exports._exceptionWithHostPort (util.js:1049:20)
13:38:16.729 gyp ERR! stack     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1085:14)
13:38:16.729 gyp ERR! System Linux 3.10.0-514.10.2.el7.x86_64
13:38:16.729 gyp ERR! command "/usr/local/node-v6.9.1-linux-x64/bin/node" "/usr/local/node-v6.9.1-linux-x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--loglevel=http" "--module=/home/xxx/xxx/xxx/node_modules/microtime/build/Release/microtime.node" "--module_name=microtime" "--module_path=/home/xxx/xxx/xxx/node_modules/microtime/build/Release"
13:38:16.729 gyp ERR! cwd /home/xxx/xxx/xxx/node_modules/microtime
13:38:16.729 gyp ERR! node -v v6.9.1
13:38:16.729 gyp ERR! node-gyp -v v3.6.0
13:38:16.729 gyp ERR! not ok 
13:38:16.729 node-pre-gyp ERR! build error 
...
```

I get timeout on IP: 104.20.22.46:443 and 192.30.253.113:443 which looks like to be cloudflare and github. Probably my corporate proxy blocking those requests.

Any suggestion other than "Fix your corporate proxy" ?

**Comment by wadey** [comment]
>>> Edit:
Why you don't publish build version of your package?

That's what this recent change is about. Testing with windows first and will do Linux next. Although you will need to fix your firewall to download the prebuilt version too as that will be hosted on github.

**Comment by faustbrian** [comment]
Any ETA when pre-builds will be available? At the moment having trouble installing a few things because microtime is interrupting.

**Comment by wadey** [comment]
@faustbrian what error are you seeing? Most users should be able to build from source on Linux/macOS.

**Comment by faustbrian** [comment]
Just running `yarn install` on macOS results in this.

```
→ yarn install
yarn install v0.24.6
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
[1/1] ⠄ microtime: http 404 status code downloading tarball https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar
[-/1] ⠄ waiting...
[-/1] ⠄ waiting...
[-/1] ⠄ waiting...
warning Error running install script for optional dependency: "/Users/Brian/Sites/my_project/node_modules/microtime: Command failed.\nExit code: 1\nCommand: sh\nArguments: -c node-pre-gyp install --fallback-to-build --loglevel http\nDirectory: /Users/Brian/Sites/my_project/node_modules/microtime\nOutput:\nnode-pre-gyp http GET https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz\nnode-pre-gyp http 404 https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz\nnode-pre-gyp ERR! Tried to download(404): https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz \nnode-pre-gyp ERR! Pre-built binaries not found for microtime@2.1.5 and node@8.1.0 (node-v57 ABI) (falling back to source compile with node-gyp) \nnode-pre-gyp http 404 status code downloading tarball https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz \nnode-pre-gyp ERR! build error \nnode-pre-gyp ERR! stack Error: Failed to execute 'node-gyp clean' (Error: spawn node-gyp ENOENT)\nnode-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/Brian/Sites/my_project/node_modules/microtime/node_modules/node-pre-gyp/lib/util/compile.js:77:29)\nnode-pre-gyp ERR! stack     at emitOne (events.js:115:13)\nnode-pre-gyp ERR! stack     at ChildProcess.emit (events.js:210:7)\nnode-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:195:12)\nnode-pre-gyp ERR! stack     at onErrorNT (internal/child_process.js:366:16)\nnode-pre-gyp ERR! stack     at _combinedTickCallback (internal/process/next_tick.js:102:11)\nnode-pre-gyp ERR! stack     at process._tickCallback (internal/process/next_tick.js:161:9)\nnode-pre-gyp ERR! System Darwin 16.6.0\nnode-pre-gyp ERR! command \"/usr/local/Cellar/node/8.1.0_1/bin/node\" \"/Users/Brian/Sites/my_project/node_modu✨  Done in 9.17s.
```

**Comment by wadey** [comment]
@faustbrian ah looks like this is an open issue with yarn and node-pre-gyp noted here: https://github.com/mapbox/node-pre-gyp/pull/292

I'll see if there is anything I can do or if I can get the Linux/macOS binaries working

**Comment by faustbrian** [comment]
`npm install` results in this.

```
→ npm install

> microtime@2.1.5 install /Users/Brian/Sites/my_project/node_modules/microtime
> node-pre-gyp install --fallback-to-build --loglevel http

node-pre-gyp http GET https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz
node-pre-gyp http 404 https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for microtime@2.1.5 and node@8.1.0 (node-v57 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v57-darwin-x64.tar.gz
  CXX(target) Release/obj.target/microtime/src/microtime.o
  SOLINK_MODULE(target) Release/microtime.node
```

**Comment by wadey** [comment]
@faustbrian looks like it works with npm (fails to download prebuilt and then build from source). So I'm pretty sure your issue is that open ticket on node-pre-gyp.

**Comment by wadey** [comment]
@faustbrian it looks like I can probably work around that issue by switching to `prebuild-install` instead of `node-pre-gyp`. It looks like `prebuild-install` is used by more big packages (like Leveldown) so I'm going to work on switching now.

**Comment by faustbrian** [comment]
@wadey Just ran `yarn install` again after always failing yesterday and seems to be working now, everything installed without any issues.

**Comment by brnrajoriya** [comment]
404 https://github.com/wadey/node-microtime/releases/download/v2.1.5/node-v48-linux-x64.tar.gz

**Comment by wadey** [comment]
I've released 2.1.6 which provides prebuilt binaries for linux and uses https://github.com/mafintosh/prebuild-install which is used by larger projects like https://github.com/Level/leveldown and seems to support `yarn` better.

Please test with 2.1.6 and re-open if you are still having issues.
